api/analyzedphenotypes.cv.api.inc

```diff
@@ -0,0 +1,547 @@
+        'width' => $info['width'],
+        'scale_to' => $options['scaleto'],
+       ));
+
+      $photo_dir = 'public://ap-photo';
+      file_prepare_directory($photo_dir, FILE_CREATE_DIRECTORY);
+
+      // Create a blank image using the computed height and width, then resample uploaded
+      // image into this blank image which will become the traits photo.
+      $new_image = imagecreatetruecolor($scale['width'], $scale['height']);
+      $img_file  = imagecreatefromjpeg($source_file);
+      imagecopyresampled($new_image, $img_file, 0, 0, 0, 0, $scale['width'], $scale['height'], $info['width'], $info['height']);
+
+      $new_filename = $identifiers['cvterm_id'] . '_' . ($i + 1) . '.jpg';
+      imagejpeg($new_image, drupal_realpath($photo_dir) . '/' . $new_filename, 100);
+
+      imagedestroy($img_file);
+      imagedestroy($new_image);
+
+      // Inform Drupal file system about this alterations.
+      // When file is present in file_managed, skip this block.
+      $check_file = entity_load('file', FALSE, array('filename' => $new_filename));
+      if (empty($check_file)) {
+        $tmp_file->filename = $new_filename;
+        $tmp_file->uri = $photo_dir . '/' . $new_filename;
+        $tmp_file->status = FILE_STATUS_PERMANENT;
+        file_save($tmp_file);
+        file_usage_add($tmp_file, 'analyzedphenotypes', 'file', 1);
+      }
+    }
+  }
+}
+
+/**
+ * FUNCTION:
+ * Dowload/get photo file information of a cvterm.
+ * Dependencies:
+ *
+ * @param $cvterm_id
+ *   An integer, controlled vocabulary id number.
+ *
+ * @param
+ *   An array, file meta data.
+ */
+function ap_download_cvtermphoto($cvterm_id) {
+  $cvtermphoto = null;
+  $photo = array('ap_photo_1', 'ap_photo_2');
+
+  foreach($photo as $i => $v) {
+    $filename = $cvterm_id . '_' . ($i + 1) . '.jpg';
+    $filedir = 'public://ap-photo';
+
+    $fileuri = drupal_realpath($filedir) . '/' . $filename;
+
+    if (file_exists($fileuri)) {
+      $info = image_get_info($fileuri);
+
+      $cvtermphoto[$v] = array(
+        'filename' => $filename,
+        'uri' => $filedir . '/' . $filename,
+        'height' => $info['height'],
+        'width' => $info['width'],
+      );
+    }
+    else {
+      // No photo.
+      $cvtermphoto[$v] = null;
+    }
+  }
+
+  return $cvtermphoto;
+}
+
+/**
+ * FUNCTION:
+ * Handle autocomplete fields that takes in cvterm name (cv name) or cvterm_id.
+ * Return value may either be cvterm_id or cvterm name (cv name).
+ * Format: cvterm name (cv name).
+ *
+ * @param $identifiers
+ *   An array with the following key:
+ *     - keyword : request terms given a keyword used in populating field suggestions/options.
+ *     or - cvtermcv_cvtermid : request cvterm id number given a cvterm name (cv name).
+ *     or - cvtermid_cvtermcv : request cvterm name (cv name) given a cvterm id number.
+ *
+ * @return
+ *   cvterm name (cv name) format or cvterm id number.
+ */
+function ap_autocomplete_cvtermfields($identifiers) {
```

Comment:
This should be separated out into three separate functions. Convention seems to be to always keep autocomplete functions separated and short.